File: Card_shuffler.py
import random
'''
def genRandInt():
    number=random.randint(1,52)
    return(number)
'''
def genUnshuffledCards():
    cardNumbers=['Ace','2','3','4','5','6','7','8','9','10','Jack','Queen','King']
    cardSuites=['Spades','Clubs','Hearts','Diamonds']
    unshuffledDeck=['','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','']
    #print(len(unshuffledDeck))
    k=0
    for i in cardSuites:
        for j in cardNumbers:
            unshuffledDeck[k]=str(j)+' of '+str(i)
            #print(k)
            k=k+1
    return(unshuffledDeck)
def shuffleDeck(deckOfCards):
    shuffledDeck=['','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','','']
    m=0
    for card in deckOfCards:
        for n in range(52):
            m=random.randint(0,51)
            if(shuffledDeck[m]==''):
                shuffledDeck[m]=card
                break
            else:
                m=random.randint(0,51)
    return(shuffledDeck)
originalDeck=genUnshuffledCards()
print('Without shuffling the deck, the results are as follows:')
for elem in originalDeck:
    print(elem)
#print(shuffleDeck(originalDeck))
shuffleddDeck=shuffleDeck(originalDeck)
#print(len(shuffleddDeck))
print('After shuffling the deck 1 time, the results are as follows:')
for ele in shuffleddDeck:
    print(ele)
shuffleddDeck=shuffleDeck(originalDeck)
print('After shuffling the deck 2 times, the results are as follows:')
for elemm in shuffleddDeck:
    print(elemm)
#print(len(shuffleddDeck))
for o in range(998):
    shuffleddDeck=shuffleDeck(originalDeck)
print('After shuffling the deck 1000 times, the results are as follows:')
for elemmm in shuffleddDeck:
    print(elemmm)

File: CardShuffler.java
import java.util.*;
public class CardShuffler{
   public static String[] genUnshuffledCards(){
      String[] cardNumbers={"Ace","2","3","4","5","6","7","8","9","10","Jack","Queen","King"};
      String[] cardSuites={"Spades","Clubs","Hearts","Diamonds"};
      String[] unshuffledDeck= new String[52];
      int k=0;
      for(int i=0;i<cardSuites.length;i++){
         for(int j=0;j<cardNumbers.length;j++){
            unshuffledDeck[k]=cardNumbers[j]+" of "+cardSuites[i];
            //System.out.println(k);
            k++;
         }
      }
      return(unshuffledDeck);
   }
   public static String[] shuffleDeck(String[] deckOfCards){
      String[] shuffledDeck={"","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","","",""};
      int m=0;
      for(int a=0;a<deckOfCards.length;a++){
         for(int n=0;n<52;n++){
            m=(int) Math.random()*52;
            if(shuffledDeck[m]==""){
               shuffledDeck[m]=deckOfCards[a];
               break;
            }
            else{
               m=(int) Math.random()*52;
            }
         }
      }
      return(shuffledDeck);
   }
   public static void main(String[] args){
      String[] originalDeck=genUnshuffledCards();
      System.out.println("Unshuffled Deck is:");
      for(int i=0;i<originalDeck.length;i++){
         System.out.println(originalDeck[i]);
      }
      String[] shuffleddDeck=shuffleDeck(originalDeck);
      //System.out.println(shuffleddDeck.length);
      System.out.println("Shuffled Deck is: ");
      for(int j=0;j<shuffleddDeck.length;j++){
         System.out.println(shuffleddDeck[j]);
      }
   }
}